Corona-Warn-App/src/test/java/de/rki/coronawarnapp/contactdiary/ui/edit/ContactDiaryEditLocationsViewModelTest.kt

```diff
@@ -0,0 +1,93 @@
+package de.rki.coronawarnapp.contactdiary.ui.edit
+
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocation
+import de.rki.coronawarnapp.contactdiary.model.toEntity
+import de.rki.coronawarnapp.contactdiary.storage.repo.ContactDiaryRepository
+import io.kotest.matchers.shouldBe
+import io.mockk.MockKAnnotations
+import io.mockk.Runs
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.just
+import kotlinx.coroutines.flow.MutableStateFlow
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import testhelpers.TestDispatcherProvider
+import testhelpers.extensions.InstantExecutorExtension
+
+@ExtendWith(InstantExecutorExtension::class)
+class ContactDiaryEditLocationsViewModelTest {
+
+    lateinit var viewModel: ContactDiaryEditLocationsViewModel
+    @MockK lateinit var contactDiaryRepository: ContactDiaryRepository
+    private val location = object : ContactDiaryLocation {
+        override val locationId = 1L
+        override var locationName = "Supermarket"
+        override val stableId = 1L
+    }
+    private val locationList = listOf(location)
+
+    @BeforeEach
+    fun setUp() {
+        MockKAnnotations.init(this)
+    }
+
+    @Test
+    fun testOnDeleteAllLocationsClick() {
```

Comment:
We have an Extension for LiveData Testing:
LiveData<>.getOrAwaitValue()

Reply:
I tried but it did not work for me